controller/src/controller.rs

```diff
@@ -203,3 +205,138 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
         }
     }
 }
+
+// Get node and BottlerocketShadow names
+#[instrument]
+fn get_associated_bottlerocketshadow_name() -> Result<String> {
+    let associated_node_name = env::var("MY_NODE_NAME").context(error::GetNodeName)?;
+    let associated_bottlerocketshadow_name = brs_name_from_node_name(&associated_node_name);
+
+    event!(
+        Level::INFO,
+        ?associated_bottlerocketshadow_name,
+        "Found associated bottlerocketshadow name."
+    );
+
+    Ok(associated_bottlerocketshadow_name)
+}
+
+/// sort shadows list which uses to determine the order of node update
+/// logic1: sort shadows by crash count
+/// logic2: move the shadow which associated bottlerocketshadow node hosts controller pod to the last
+#[instrument(skip())]
+fn sort_shadows(
+    mut shadows: Vec<BottlerocketShadow>,
+    associated_brs_name: &str,
+) -> Result<Vec<BottlerocketShadow>> {
+    let mut associated_brs_node_position: usize = usize::MAX;
+    // sort shadows by crash count
+    shadows.sort_by(|a, b| a.compare_crash_count(b));
+
+    // move the shadow which associated bottlerocketshadow node hosts controller pod to the last
+    for (position, brs) in shadows.iter().enumerate() {
```

Comment:
nit: Consider using `shadows.iter().position()`. This would also allow you to use an `Option` rather than having to use `usize::MAX` as a sentinel value.

Reply:
Oh, i just remembered why I didn't use `shadows.iter().position()`. When I was trying to do like this, the position function showed it was unable to handle result value as input. Actually the position function can't handle ` result` which is `brs.metadata.name.as_ref().context(error::MissingBottlerocketShadowName)?`
```
let associated_brs_node_position = shadows.iter().position(|brs| brs.metadata.name.as_ref().context(error::MissingBottlerocketShadowName)? == associated_brs_name).context("")?
```

Reply:
Since you have `kube::ResourceExt` imported, you could just do `brs.name()` which technically panics if the name is missing. Not great, but there are likely bigger problems if the name is missing from objects.

Alternatively, if the name is missing you can probably skip the item rather than returning early:

```rust
let associated_brs_node_position = shadows
    .iter()
    .position(|brs| brs.metadata.name.as_ref() == Some(associated_brs_name))
```